include/nbl/asset/filters/CNormalMapToDerivativeFilter.h

```diff
@@ -0,0 +1,275 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef __NBL_ASSET_C_NORMAL_MAP_TO_DERIVATIVE_FILTER_H_INCLUDED__
+#define __NBL_ASSET_C_NORMAL_MAP_TO_DERIVATIVE_FILTER_H_INCLUDED__
+
+#include "nbl/core/core.h"
+
+#include <type_traits>
+#include <functional>
+ 
+#include "nbl/asset/filters/CMatchedSizeInOutImageFilterCommon.h"
+#include "CConvertFormatImageFilter.h"
+
+namespace nbl
+{
+namespace asset
+{
+
+class CNormalMapToDerivativeFilterBase
+{
+	public:
+		class CNormalMapToDerivativeStateBase
+		{
+			public:
+
+				static inline constexpr size_t decodeTypeByteSize = sizeof(double);
+				static inline constexpr size_t forcedScratchChannelAmount = 4;
```

Comment:
normal map has 3 channels

Reply:
actually scratch only needs to store 2 channels
https://github.com/Devsh-Graphics-Programming/Nabla/pull/9/files/a1ab07b2681a4d31fa8e94472fef225524b92f04..c23cd7a68eb40f58808a9d38d879ff29d8e74150#r599332402

Reply:
corrected in following commit